lib/client/api.go

```diff
@@ -2021,3 +2071,39 @@ func ParsePortForwardSpec(spec []string) (ports ForwardedPorts, err error) {
 	}
 	return ports, nil
 }
+
+// ToStringSpec returns the same string spec which can be parsed by
+// ParseDynamicPortForwardSpec.
+func (fp DynamicForwardedPorts) ToStringSpec() (retval []string) {
+	for _, p := range fp {
+		retval = append(retval, p.ToString())
+	}
+	return retval
+}
+
+// ParseDynamicPortForwardSpec parses parameter to -D flag, i.e. strings like "remote.host:3000"
+// The opposite of this function (spec generation) is DynamicForwardPorts.ToStringSpec.
+func ParseDynamicPortForwardSpec(spec []string) (ports DynamicForwardedPorts, err error) {
+	if len(spec) == 0 {
+		return ports, nil
+	}
+	const errTemplate = "Invalid dynamic port forwarding spec: '%s'. Could be like `local.host:8080`"
+	ports = make(DynamicForwardedPorts, len(spec), len(spec))
+
+	for i, str := range spec {
+		parts := strings.Split(str, ":")
+		if len(parts) > 2 {
+			return nil, fmt.Errorf(errTemplate, str)
+		}
+		if len(parts) == 1 {
+			parts = append([]string{"127.0.0.1"}, parts...)
+		}
+		p := &ports[i]
```

Comment:
why pointer here? This function needs further cleanup and should be adjusted to our style.